diff --git a/lib/plugins/db.rb b/lib/plugins/db.rb
index b0381cb..b308a78 100644
--- a/lib/plugins/db.rb
+++ b/lib/plugins/db.rb
@@ -1,4 +1,36 @@
 # -*- coding: utf-8 -*-
+
+# for Ruby 1.8
+unless String.public_method_defined?(:force_encoding)
+
+  class String
+    def force_encoding(enc)
+      self
+    end
+    def encoding
+      self
+    end
+    def encode(enc)
+      self
+    end
+  end
+
+  class Encoding
+    def self.find(str)
+      p str
+      Encoding.new
+    end
+    ASCII_8BIT = Encoding.new
+    UTF_8 = Encoding.new
+    UTF_16LE = Encoding.new
+    UTF_16BE = Encoding.new
+    def encoding(e)
+      self
+    end
+  end
+
+end
+
 require 'sequel'
 
 config.plugins.db.set_default(:path, Termtter::CONF_DIR + '/termtter.db')
