diff --git a/lib/plugins/another_stdout.rb b/lib/plugins/another_stdout.rb
index d99d489..c0c7ff2 100755
--- a/lib/plugins/another_stdout.rb
+++ b/lib/plugins/another_stdout.rb
@@ -85,7 +85,7 @@ module Termtter
     def status_line(s, time_format, event, indent = 0)
       return [] unless s
       text = TermColor.escape(s.text)
-      color = string_color(s.user.name)
+      color = string_color(s.user.screen_name)
       status_id = Termtter::Client.data_to_typable_id(s.id)
       reply_to_status_id =
         if s.in_reply_to_status_id.nil?
@@ -144,7 +144,7 @@ module Termtter
 
     def colorize_users(text, prefix = '@')
       text.gsub(/#{prefix}(\w+)/) do |i|
-        color = string_color(i)
+        color = string_color(i[1..-1])
         "<#{color}>#{i}</#{color}>"
       end
     end
