diff --git a/lib/plugins/friends.rb b/lib/plugins/friends.rb
index 46b24b1..20d3bc7 100644
--- a/lib/plugins/friends.rb
+++ b/lib/plugins/friends.rb
@@ -29,14 +29,16 @@ module Termtter::Client
     :name => :diff_follow, :aliases => [:diff],
     :exec_proc => lambda {|arg|
       user_name = arg.empty? ? config.user_name : arg
+      puts "friends:"
+      call_commands "friends" if public_storage[:friends].nil?
+      puts
+      puts "followers:"
+      call_commands "followers" if public_storage[:followers].nil?
       friends = public_storage[:friends]
       followers = public_storage[:followers]
-      if friends.nil? || followers.nil?
-        puts 'Do followers and friends first.'
-        return
-      end
       friends = friends.map(&:screen_name)
       followers = followers.map(&:screen_name)
+      puts
       puts "friends - followers:"
       puts (friends - followers).map{|s|"http://#{config.host}/#{s}"}.join("\n")
       puts
