diff --git a/lib/plugins/gsub.rb b/lib/plugins/gsub.rb
index b782d5d..23299fe 100644
--- a/lib/plugins/gsub.rb
+++ b/lib/plugins/gsub.rb
@@ -10,10 +10,8 @@ Termtter::Client.register_hook(
   :point => :filter_for_output,
   :exec => lambda {|statuses, event|
     statuses.each do |s|
-      next unless s.methods.member?('text=')
       t = s.text
       config.plugins.gsub.table.each {|a, b| t.gsub!(a, b || '') }
-      s.text = t
     end
   }
 )
