diff --git a/lib/plugins/reduce_text.rb b/lib/plugins/reduce_text.rb
index 3ec8538..a2c6781 100644
--- a/lib/plugins/reduce_text.rb
+++ b/lib/plugins/reduce_text.rb
@@ -8,12 +8,7 @@ module Termtter::Client
     :name => :reduce_screenname,
     :point => :prepare_screenname,
     :exec => lambda {|n, event|
-      if @last_screen_name == n
-        ''
-      else
-        @last_screen_name = n
-        "#{n}: "
-      end
+      @last_screen_name == n ? '' : @last_screen_name = n
     }
   )
 
