diff --git a/spec/termtter/client_spec.rb b/spec/termtter/client_spec.rb
index 52b7dcb..2062926 100644
--- a/spec/termtter/client_spec.rb
+++ b/spec/termtter/client_spec.rb
@@ -287,12 +287,7 @@ module Termtter
       Client.run
     end
 
-    it 'load_config' do
-      Client.should_receive(:legacy_config_support)
-      File.should_receive(:exist?).twice.and_return(true)
-      require 'termtter/config_setup'
-      Client.load_config
-    end
+    it 'load_config'
 
     it 'does nothing when ~/.termtter is directory' do
       File.should_receive(:ftype).and_return('directory')
