diff --git a/lib/plugins/another_stdout.rb b/lib/plugins/another_stdout.rb
index c4f2f8a..9f6eaa3 100755
--- a/lib/plugins/another_stdout.rb
+++ b/lib/plugins/another_stdout.rb
@@ -92,6 +92,13 @@ module Termtter
           Termtter::Client.data_to_typable_id(s.in_reply_to_status_id)
         end
 
+      retweeted_status_id =
+        if s.retweeted_status
+          Termtter::Client.data_to_typable_id(s.retweeted_status.id)
+        else
+          nil
+        end
+
       time = "#{Time.parse(s.created_at).strftime(time_format)}"
       time = Client.get_hooks(:prepare_time).inject(time){|result, hook| hook.call(result, event)}
 
@@ -119,12 +126,35 @@ module Termtter
 
       texts = [erbed_texts]
 
-      if config.plugins.stdout.show_as_thread && s.in_reply_to_status_id
-        texts.concat status_line(Status[s.in_reply_to_status_id], time_format, event, indent + 1)
+      if config.plugins.stdout.show_reply_chain && s.in_reply_to_status_id
+        indent += 1
+        unless indent > config.plugins.stdout.max_indent_level
+          begin
+            status = Termtter::API.twitter.show(s.in_reply_to_status_id)
+            texts.concat status_line(status, time_format, event, indent)
+          rescue Rubytter::APIError
+          end
+        end
       end
 
       texts
     end
+
+    def colorize_users(text)
+      text.gsub(/@(\w+)/) do |i|
+        user = Termtter::API.twitter.cached_user($1)
+        if user
+          color = user_color(user)
+          "<#{color}>#{i}</#{color}>"
+        else
+          i
+        end
+      end
+    end
+
+    def user_color(user)
+      config.plugins.stdout.colors[user.id.to_i % config.plugins.stdout.colors.size]
+    end
   end
 
   Client.register_hook(AnotherStdOut.new)
