diff --git a/lib/plugins/stream.rb b/lib/plugins/stream.rb
index 079bf19..6768c9e 100644
--- a/lib/plugins/stream.rb
+++ b/lib/plugins/stream.rb
@@ -10,14 +10,14 @@ module Termtter::Client
 
   class << self
     if defined?(DB)
-      def friends(max = 999999)
+      def friends(max)
         Status.group(:user_id).
           select(:user_id, :screen_name).
           join(:users, :id => :user_id).
           order(:COUNT.sql_function.desc).take(max)
       end
     else
-      def friends(max = 1/0.0)
+      def friends(max)
         friends = []
         page    = 0
         begin
@@ -51,7 +53,6 @@ module Termtter::Client
         TweetStream::Client.new(config.user_name, config.password).
           filter(:track => args) do |status|
           output [Termtter::ActiveRubytter.new(status)], :update_friends_timeline
-          Readline.refresh_line
         end
       end
     end
@@ -101,7 +102,6 @@ module Termtter::Client
           TweetStream::Client.new(config.user_name, config.password).
             filter(:follow => current_targets) do |status|
             output [Termtter::ActiveRubytter.new(status)], :update_friends_timeline
-            Readline.refresh_line
           end
         rescue(NoMethodError) => e    # #<NoMethodError: private method `split' called for nil:NilClass>
           puts "stream seems broken (#{e.inspect})."
