diff --git a/lib/plugins/defaults/standard_commands.rb b/lib/plugins/defaults/standard_commands.rb
index f37d701..413e3ef 100644
--- a/lib/plugins/defaults/standard_commands.rb
+++ b/lib/plugins/defaults/standard_commands.rb
@@ -28,7 +28,7 @@ module Termtter::Client
   register_command(
     :name => :update, :alias => :u,
     :exec => lambda {|arg|
-      unless arg.empty?
+      unless arg.rstrip.empty?
         params =
           if config.easy_reply && /^\s*(@\w+)/ =~ arg
             user_name = normalize_as_user_name($1)
